components/Chips/src/MDCChipView.m

```diff
@@ -431,10 +450,12 @@ - (void)layoutSubviews {
 
   _selectedImageView.alpha = self.showSelectedImageView ? 1 : 0;
 
-  CGFloat cornerRadius = MIN(CGRectGetHeight(self.frame), CGRectGetWidth(self.frame)) / 2;
-  self.layer.cornerRadius = cornerRadius;
-  self.layer.shadowPath =
-      [UIBezierPath bezierPathWithRoundedRect:self.bounds cornerRadius:cornerRadius].CGPath;
+  if (!self.layer.shapeGenerator) {
+    CGFloat cornerRadius = MIN(CGRectGetHeight(self.frame), CGRectGetWidth(self.frame)) / 2;
+    self.layer.cornerRadius = cornerRadius;
+    self.layer.shadowPath =
+        [UIBezierPath bezierPathWithRoundedRect:self.bounds cornerRadius:cornerRadius].CGPath;
```

Comment:
Do we need to reset these values if the shape generator is set?

Reply:
Done.